integration-tests/test_connection.py

```diff
@@ -6,13 +6,31 @@
 :upstream: Yes
 """
 
+import os
+import typing
+
 import pytest
 
+if typing.TYPE_CHECKING:
+    import pytest_client_tools.insights_client
+
 # the tests need a valid connection to insights so therefore the subman registration
 pytestmark = pytest.mark.usefixtures("register_subman")
 
 
-def test_connection(insights_client):
+def _is_using_proxy(
+    config: "pytest_client_tools.insights_client.InsightsClientConfig",
+) -> bool:
+    if os.getenv("HTTP_PROXY") is not None:
```

Comment:
eh, I was setting a proxy in Unix would be this easy... I think it should be better to also check for `http_proxy` (lowercase) which (according to my experience) should be more commonly used

... and actually insights-core does not use `HTTP_PROXY` (which explicitly rejects), but `HTTPS_PROXY`:

https://github.com/RedHatInsights/insights-core/blob/23157b137ffa6ca3ee9f0d12ecf13e23f5f4a80d/insights/client/config.py#L553-L561

also, please keep into account that the proxy can be configured also in `rhsm.conf`, in case of autoconfig (which is true by default), so most likely you need to check that too, for added fun...

https://github.com/RedHatInsights/insights-core/blob/23157b137ffa6ca3ee9f0d12ecf13e23f5f4a80d/insights/client/connection.py#L334-L339